### PR TITLE
typecheck trio/_core/_unbounded_queue

### PR DESCRIPTION
**Pull request by jakkdl**

### CODE REVIEW

**Review by TeamSpen210 — APPROVED** [review]
Looks good to me, a straightforward one.

File: trio/_core/_unbounded_queue.py [hunk 1]
```diff
@@ -41,26 +60,27 @@ class UnboundedQueue(metaclass=Final):
 
     """
 
+    # deprecated is not typed
```
Comment (A5rocks) [inline trio/_core/_unbounded_queue.py:63]: This isn't true anymore

File: trio/_core/_unbounded_queue.py [hunk 2]
```diff
@@ -128,22 +148,14 @@ async def get_batch(self):
...
-        Currently the following fields are defined:
-
-        * ``qsize``: The number of items currently in the queue.
-        * ``tasks_waiting``: The number of tasks blocked on this queue's
-          :meth:`get_batch` method.
-
-        """
-        return _UnboundedQueueStats(
+    def statistics(self) -> UnboundedQueueStatistics:
+        """Return an UnboundedQueueStatistics object containing debugging information."""
```
Comment (A5rocks) [inline trio/_core/_unbounded_queue.py:152]:
```suggestion
        """Return an `UnboundedQueueStatistics` object containing debugging information."""
```

... maybe that's nicer?

**Review by A5rocks — CHANGES_REQUESTED** [review]
Probably just have to merge in master

File: trio/_core/_unbounded_queue.py (hunk 2 above)
Comment (jakkdl) [reply trio/_core/_unbounded_queue.py:151]: Or ```""":class:`UnboundedQueueStatistics`"""```, though as it's deprecated and not documented anywhere it doesn't matter all that much.

**Pull request merged**